panc/src/main/java/org/quattor/pan/dml/data/ProtectedListResource.java

```diff
@@ -39,13 +43,28 @@ public Element put(int index, Element newValue) {
 	}
 
 	@Override
+	public void append(Element e) {
+		throw CompilerError.create(MSG_ILLEGAL_WRITE_TO_PROTECTED_LIST);
+	}
+
+	@Override
+	public void prepend(Element e) {
+		throw CompilerError.create(MSG_ILLEGAL_WRITE_TO_PROTECTED_LIST);
+	}
+
+	@Override
 	public int size() {
 		return baseList.size();
 	}
 
 	@Override
 	public Resource.Iterator iterator() {
-		return baseList.iterator();
+		return baseList.protectedIterator();
```

Comment:
so this is similar: you are not protecting the elements in the underlying list, you are always returning `.protect()` copies of an unprotected underlying list. is there any reason why the underlying elements could not be made a protectedresource?